CMakeLists.txt

```diff
@@ -187,3 +187,14 @@ add_custom_target(check-stablehlo-quick)
 add_dependencies(check-stablehlo-ci check-stablehlo-quick)
 
 add_subdirectory(stablehlo)
+
+# Let's create a single library here so that linking against
+add_library(stablehlo_lib INTERFACE)
+# Link the master library with the other libraries
+target_link_libraries(stablehlo_lib INTERFACE
+  ${dialect_libs}
+  ${conversion_libs}
+  StablehloOps
+  )
+
+add_subdirectory(examples)
```

Comment:
@sdasgup3 are you familiar enough with CMake to know if this pattern is a good idea?
The goal would be to make a single target library (interface) that includes everything we need to make examples easier to work with.

Reply:
Not super familiar but I can see similar pattern in other places like https://github.com/openxla/stablehlo/blob/04365f85cfbffe3d95ba2fb79ff34cd929d4a9a6/stablehlo/tools/CMakeLists.txt#L49.

A few nits: 
1. Can we home this lib in examples/c++ directory instead, where it is actually consumed. 
2. Name it `stablehlo-example-lib` (with dashes) to be  consistent with other lib target names.